hack/build.sh

```diff
@@ -2,6 +2,16 @@
 
 set -ex
 
+version() { IFS="."; printf "%03d%03d%03d\\n" $@; }
```

Comment:
It looks like IFS is globally set to `.` and never reverted to what it originally was (perhaps unset). Is it possible that this is causing the failures?

Reply:
Perhaps easiest to have `local IFS="."` and this should avoid any issues with IFS after function is run. But this is not causing the issue (sorry I didn't read issue description initially).

We can't quote `$@` because it is passed to function as a single parameter and quoting will prevent splitting on IFS. But shellcheck can't detect automatically whether we are deliberately omitting quotes or are just idiots. So I guess we can prepend fuction with `# shellcheck disable=SC2068` to avoid the error.

I'm wondering whether `$@` is the best idea here. Maybe `$1` will make more sense. In either case adding the appropriate directive (`SC2086` for `$1`) should get us going.

Reply:
This should work I think.
```
# shellcheck disable=SC2086
version() { local IFS="."; printf "%03d%03d%03d\\n" $1; }
version 1.8.1
```

Reply:
sorry for the spam. Now it complains that `sh` does not support `local` declaration :disappointed: 
This returns no errors and works for me:
```
#!/bin/sh
# shellcheck disable=SC2086 disable=SC2183
version() { IFS="."; printf "%03d%03d%03d\\n" $1; unset IFS; }
version 1.8.1
```

Reply:
That means that we have to change the shellcheck script?

Reply:
@akostadinov Thanks. I didn't know that we can disable specific check codes for shellcheck.

Reply:
I think `unset IFS;` is important because setting variables in functions in sh is global IIRC. So setting but not reverting the variable is potentially dangerous for the rest of the script.